### PR TITLE
Add ElevenLabs to Docs

### PR DESCRIPTION
**Pull request by PineappleIOnic**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 1]
```diff
@@ -0,0 +1,358 @@
+---
+layout: article
+title: Integrating ElevenLabs into Appwrite
+description: Learn how to integrate ElevenLabs into your Appwrite project.
+difficulty: beginner
+readtime: 15
+back: /docs/products/ai/audio-processing
+---
+
+Integrating ElevenLabs into your Appwrite project is simple. This tutorial will guide you through the process of setting up the ElevenLabs API and integrating it into your Appwrite project.
```
Comment (christyjacob4) [inline src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc:10]: Screenshot 2024-03-27 at 12 46 58 PM

This paragraph should talk about what eleven labs does.

File: static/images/docs/ai/integrations/elevenlabs/demo.png
Comment (christyjacob4) [inline static/images/docs/ai/integrations/elevenlabs/demo.png:1]: What about the dark mode version of this image?

Comment (PineappleIOnic) [reply static/images/docs/ai/integrations/elevenlabs/demo.png:1]: To keep the demo simple I didn't add any theming support to it

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+![Create function screen](/images/docs/functions/dark/template.png)
+{% /only_dark %}
+{% only_light %}
+![Create function screen](/images/docs/functions/template.png)
+{% /only_light %}
+
+1. In the Appwrite Console's sidebar, click **Functions**.
+2. Click **Create function**.
+3. Under **Connect Git repository**, select your provider.
+4. After connecting to GitHub, under **Quick start**, select the **Node.js** starter template.
```
Comment (christyjacob4) [inline src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc:29]: Node Js starter template? This should be eleven labs right ?

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+5. In the **Variables** step, tick the box to **Generate API key on completion**.
+6. Follow the step-by-step wizard and create the function.
+{% /section %}
+
+{% section #step-2 step=2 title="Add Fetch" %}
+Once your function is created, navigate to the freshly created repository and either clone it to your local machine or launch a GitHub Codespace.
```
Comment (christyjacob4) [inline src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc:35]: > or launch a GitHub Codespace. 

Is this how what we suggest in other function guides as well? Else lets remove it

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+
+Since we're going to be making requests to the ElevenLabs API and fetch isn't included in the current NodeJS 18 Runtime, we're going to install the `node-fetch` package.
+We're also going to install the `node-appwrite` package so we can upload the generated audio file to Appwrite Storage.
+```bash
+npm install node-fetch node-appwrite
+```
+{% /section %}
+
+{% section #step-3 step=3 title="Create utility functions" %}
+For our example, our function will be able to take both `GET` and `POST` requests.
```
Comment (christyjacob4) [inline src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc:45]: As discussed, lets avoid references to first person. So skip words like we, I, our etc.

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+
+For the `GET` request we will return a static HTML page we'll write later that will use AlpineJS to make a `POST` request to our function and the `POST` request will return the ElevenLabs audio stored in Appwrite Storage.
```
Comment (christyjacob4) [inline src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc:47]: This sentence is too dense. Its better to split it into two

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+ * Returns the contents of a file in the static folder
+ * @param {string} fileName
+ * @returns {string} Contents of static/{fileName}
+ */
+export function getStaticFile(fileName) {
+  return fs.readFileSync(path.join(staticFolder, fileName)).toString();
+}
+```
+
+By separating the `getStaticFile()` function into a `utils.js` file, we can easily reuse this function in other parts of our function if we extend on top of this.
```
Comment (christyjacob4) [inline src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc:70]: I feel this sentence may not be required.. its not adding any additional value. Throughout the post, try to trim down on lines that are not required and keep it as short and concise as possible  

>  we can easily reuse this function in other parts of our function
Also this doesn't sound right.

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+
+While we're at it, we're also going to create a `throwIfMissing()` function in the same file which we'll use later to validate body and environment variables:
```
Comment (christyjacob4) [inline src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc:72]:
```suggestion
Similarly, create a `throwIfMissing()` function in the same file that will be used later to validate environment variables and the body. 
```
Try to re-write all the sentences with the above tone

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+  }
+  if (missing.length > 0) {
+    throw new Error(`Missing required fields: ${missing.join(', ')}`);
+  }
+}
+```
+
+{% /section %}
+
+{% section #step-4 step=4 title="Create HTML File" %}
```
Comment (christyjacob4) [inline src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc:96]: Maybe its better to say "Create Landing Page" or something

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc (hunk 2 above)
Comment (PineappleIOnic) [reply src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc:31]: Not if they are creating this from scratch

**Review by gewenyu99 — APPROVED** [review]

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc (hunk 3 above)
Comment (PineappleIOnic) [reply src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc]: Removed

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 9]
```diff
@@ -0,0 +1,358 @@
...
+4. After connecting to GitHub, under **Quick start**, select the **Node.js** starter template.
+5. In the **Variables** step, tick the box to **Generate API key on completion**.
+6. Follow the step-by-step wizard and create the function.
+{% /section %}
+
+{% section #step-2 step=2 title="Add Fetch" %}
+Once your function is created, navigate to the freshly created repository and either clone it to your local machine or launch a GitHub Codespace.
+
+Since we're going to be making requests to the ElevenLabs API and fetch isn't included in the current NodeJS 18 Runtime, we're going to install the `node-fetch` package.
+We're also going to install the `node-appwrite` package so we can upload the generated audio file to Appwrite Storage.
```
[the remark this answers is not in the document]
Comment (PineappleIOnic) [reply src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc]: Merged

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc (hunk 4 above)
Comment (PineappleIOnic) [reply src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc]: Done

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc (hunk 5 above)
Comment (PineappleIOnic) [reply src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc]: Done

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc (hunk 6 above)
Comment (PineappleIOnic) [reply src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc]: Alright, removed

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 10]
```diff
@@ -0,0 +1,358 @@
...
+
+{% section #step-10 step=10 title="Test our function" %}
+Now that our function is deployed, we can test it by visiting the function URL in our browser.
+This should show the UI we created earlier and to test it we can write a prompt and click the submit button, after a brief moment you should see the audio appear below the input.
+
+![Testing the function](/images/docs/ai/integrations/elevenlabs/demo.png)
+{% /section %}
+
+{% section #step-11 step=11 title="Conclusion" %}
+In this tutorial, we've created a simple web page that allows us to interact with the ElevenLabs API using Appwrite Functions. We've learned how to create a function that interacts with an external API, upload audio to storage, how to deploy it to Appwrite, and how to create an interactive web page that uses the function.
```
[the remark this answers is not in the document]
Comment (PineappleIOnic) [reply src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc]: I'll keep it in for now as it matches all the others, we can go through all them together once merged to ensure they all have the same standards

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 11]
```diff
@@ -0,0 +1,358 @@
...
+
+
+We look forward to seeing what you build with Appwrite and ElevenLabs!
```
[the remark this answers is not in the document]
Comment (PineappleIOnic) [reply src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc]: Alright, Removed

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc (hunk 2 above)
Comment (christyjacob4) [reply src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc:31]: but dont you have a template for eleven labs as well ?

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 12]
```diff
@@ -4,5 +4,319 @@ title: Integrating ElevenLabs into Appwrite
...
+    });
+  }
+};
+```
+
+A check is also included to ensure that the `ELEVENLABS_API_KEY`, `APPWRITE_API_KEY` and `APPWRITE_BUCKET_ID` environment variables is set.
+{% /section %}
+
+{% section #step-5 step=5 title="Create web page" %}
+Now create a HTML web page that the function will serve. Create a new `static` folder in the root of your project and add an `index.html` with some html boilerplate:
```
Comment (christyjacob4) [inline src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc:120]:
```suggestion
Now create a HTML web page that the function will serve. Create a new `static` folder in the root of your project and add an `index.html` with some HTML boilerplate:
```

File: src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc [hunk 13]
```diff
@@ -4,5 +4,319 @@ title: Integrating ElevenLabs into Appwrite
...
+{% section #step-8 step=8 title="Test the function" %}
+Now that the function is deployed, test it by visiting the function URL in your browser.
+This should show the UI created earlier and to test it, write a prompt and click the submit button. After a brief moment you should see the audio appear below the input.
+
+![Testing the function](/images/docs/ai/integrations/elevenlabs/demo.png)
+{% /section %}
+
+{% section #step-9 step=9 title="Next steps" %}
+In this tutorial, you've created a simple web page that allows you to interact with the ElevenLabs API using Appwrite Functions. You've learned how to create a function that interacts with an external API, upload audio to storage, how to deploy it to Appwrite, and how to create an interactive web page that uses the function.
+{% /section %}
```
Comment (christyjacob4) [inline src/routes/docs/products/ai/integrations/elevenlabs/+page.markdoc:322]: This section is not required. We aren't really giving any next steps here. Lets skip it.

**Pull request merged**